### PR TITLE
Feat/predev scripts (#38)

### PR DESCRIPTION
**Pull request by ReiniervanLimpt**
What do you guys think about these changes to the gulpfile and package, ive included:

1. a predev script which runs build:css
2. a prebuild:css script which removes the dist/src folder so build:css can create a new one from he css file in `src/scss/*`
3. a watch:css script which can be run in a seperate terminal `npm run watch:css` which proxies our port from localhost:2999 to localhost:3000, it watches the following directories and files for changes: 
```javascript
  gulp.watch('./src/**/*.scss', gulp.series('css'))
  gulp.watch('./views/*.hbs', browserSync.reload)
```
if any changes are made to files in these directories browser-sync instantly refreshes your proxied page (localhost:3000).

Ive also removed a script which called for "gulp" which is not a specified task in the gulpfile, do you need this script Daniel?

Anyway its not a necessary change but its one i made because the css in dist needs a script to be built and the watch:css script is very handy when you want to make css changes!

In summary:

- included a predev script which builds our css from our sourcemap
- included a watch:css script with browser-sync to automatically refresh your localhost when you make changes to hbs or scss files

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
+require("dotenv").config()
 const gulp = require("gulp")
 const sass = require("gulp-sass")
 const autoprefixer = require("gulp-autoprefixer")
 const sourcemaps = require("gulp-sourcemaps")
 const uglify = require("gulp-uglify")
 const concat = require("gulp-concat")
 const browserSync = require('browser-sync').create()
-const port = 2999
+const port = process.env.PORT
```
Comment (DanielvandeVelde) [inline gulpfile.js:9]: We can change this to: 
```js 
const port = process.env.PORT || 3000
```
Setting up via the .env variables but also providing a back-up just in case.

File: package.json [hunk 2]
```diff
@@ -40,8 +42,11 @@
     "node": "^15.12.0"
   },
   "scripts": {
+    "predev": "npm run build:css",
     "dev": "nodemon index.js",
-    "build": "gulp",
```
Comment (DanielvandeVelde) [inline package.json:44]: The `"build": "gulp"`-script runs the default in the gulpfile.  
In our case this is:
```js
gulp.task("default", gulp.parallel("css", "js"))
```
By removing it I think you removed this css and js build-step 😄 
I think you re-added the CSS somewhere, but not the JS?
As long as you make sure that if you delete the dist/js && css folder everything still builds there, we're fine.

Comment (ReiniervanLimpt) [reply package.json:44]: ahh i see, running the gulp command did nothing for me in my local environment, ill make sure to include it

Comment (ReiniervanLimpt) [reply package.json:44]: just kidding, it did... ill fix this soon

**Comment by ReiniervanLimpt** [comment]
Ive made changes to the script chain (included the default gulp task again :stuck_out_tongue_winking_eye: 
Checkt he latest commit.

File: gulpfile.js (hunk 1 above)
Comment (deannabosschert) [reply gulpfile.js:9]: I'm with Daniel on this.
I also use the || when it comes to the port, plus I like 3000 more than 2999

**Review by deannabosschert — APPROVED** [review]
Looks great! I'm with Daniel on the port thing, but it's not a necessity to fix for this PR

I'm also pleased to see the rimraf;  I usually use this for static sites but I think it's clever to have found a way in useful implementation.
